src/core/collision_detection/BindAtPointOfCollision.hpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2011-2024 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <config/config.hpp>
+
+#ifdef COLLISION_DETECTION
+#ifdef VIRTUAL_SITES_RELATIVE
+
+#include "CollisionPair.hpp"
+#include "common.hpp"
+
+#include "Particle.hpp"
+#include "system/System.hpp"
+
+#include <vector>
+
+namespace CollisionDetection {
+
+class BindAtPointOfCollision {
+public:
+  double distance;
+  // Square of distance at which particle are bound
+  double distance_sq;
+
+  /// bond type used between centers of colliding particles
+  int bond_centers;
+  /// bond type used between virtual sites
+  int bond_vs;
+  /** Placement of virtual sites for MODE_VS.
+   *  0=on same particle as related to,
+   *  1=on collision partner,
+   *  0.5=in the middle between
+   */
+  double vs_placement;
+  /// particle type for virtual sites created on collision
+  int part_type_vs;
+
+  BindAtPointOfCollision(double distance, int bond_centers, int bond_vs,
+                         double vs_placement, int part_type_vs)
+      : distance{distance}, distance_sq{distance * distance},
+        bond_centers{bond_centers}, bond_vs{bond_vs},
+        vs_placement{vs_placement}, part_type_vs{part_type_vs} {}
+
+  void initialize(System::System &system);
+
+  auto cutoff() const { return distance; }
+
+  void handle_collisions(System::System &system,
+                         std::vector<CollisionPair> &local_collision_queue);
+
+  bool detect_collision(Particle const &p1, Particle const &p2,
+                        double const dist_sq) const {
+    if (dist_sq > distance_sq)
+      return false;
```

Comment:
We thought about it, but `GlueToSurface` has an extra condition that is cheaper to evaluate than `detect_collision_common()`, yet more expensive to evaluate than the distance check.

It is also debatable whether this distance check should be performed here, or rather in the collision detection class, since it would save a function call. In the end we opted to leave the distance check inside the protocols, since we are not relying on polymorphism, and thus function calls are cheap. This design also allows us to explore protocols that don't rely on a distance check, like Russel's phase simulations that depend on particle orientation.